e2e/tests/negative/node_down.robot

```diff
@@ -0,0 +1,145 @@
+
+Power Off Node And Force Delete Terminating Statefulset Pod
+    [Arguments]    ${node_down_pod_deletion_policy}
+    Given Set setting default-replica-count to 2
+    And Set setting node-down-pod-deletion-policy to ${node_down_pod_deletion_policy}
+    And Create storageclass longhorn-test with    dataEngine=${DATA_ENGINE}
+
+    When Create statefulset 0 using RWO volume with longhorn-test storageclass
+    Then Write 100 MB data to file data in statefulset 0
+    And Power off volume node of statefulset 0
+    And Sleep    360
+    And Check Longhorn node READY state on power off node is False
+    And Wait for statefulset 0 pod is Running on another node
+    And Wait for statefulset 0 pods stable
+    And Check statefulset 0 data in file data is intact
+
+    When Power on off nodes
+    And Wait for longhorn ready
+
+Power Off Node And Not Force Delete Terminating Deployment Pod
+    [Arguments]    ${node_down_pod_deletion_policy}
+    Given Set setting default-replica-count to 2
+    And Set setting node-down-pod-deletion-policy to ${node_down_pod_deletion_policy}
+    And Create storageclass longhorn-test with    dataEngine=${DATA_ENGINE}    numberOfReplicas=2
+
+    When Create persistentvolumeclaim 0 using RWO volume with longhorn-test storageclass
+    And Create deployment 0 with persistentvolumeclaim 0
+    Then Write 100 MB data to file data in deployment 0
+    And Power off volume node of deployment 0
+    And Sleep    360
+    And Check Longhorn node READY state on power off node is False
+    And Wait for deployment 0 pod stuck in Terminating on the original node
+    And Wait for deployment 0 pod stuck In ContainerCreating on another node
+
+    When Force delete deployment 0 pod on the original node
+    And Wait for deployment 0 pod is Running on another node
+    And Wait for deployment 0 pods stable
+    And Check deployment 0 data in file data is intact
+
+    When Power on off nodes
+    And Wait for longhorn ready
+
+Power Off Node And Force Delete Terminating Deployment Pod
+    [Arguments]    ${node_down_pod_deletion_policy}
+    Given Set setting default-replica-count to 2
+    And Set setting node-down-pod-deletion-policy to ${node_down_pod_deletion_policy}
+    And Create storageclass longhorn-test with    dataEngine=${DATA_ENGINE}    numberOfReplicas=2
+
+    When Create persistentvolumeclaim 0 using RWO volume with longhorn-test storageclass
+    And Create deployment 0 with persistentvolumeclaim 0
+    Then Write 100 MB data to file data in deployment 0
+    And Power off volume node of deployment 0
+    And Sleep    360
+    And Check Longhorn node READY state on power off node is False
+    And Wait for deployment 0 pod is Running on another node
+    And Wait for deployment 0 pods stable
+    And Check deployment 0 data in file data is intact
+
+    When Power on off nodes
+    And Wait for longhorn ready
+
+*** Test Cases ***
+Physical Node Down StatefulSet Deletion Policy Do Nothing
```

Comment:
NIT: Consider consolidating the test cases that are using the same template. Example: https://github.com/longhorn/longhorn-tests/blob/c5c9387ca357592fa12f90cd4eda1b46b77675c6/e2e/tests/negative/node_reboot_with_different_anti_affinity.robot#L57-L80

Reply:
Updated to below format

```robot
*** Test Cases ***    DESCRIPTION    NODE-DOWN-POD-DELETION-POLICY
Physical Node Down StatefulSet Deletion Policy Do Nothing
    [Tags]    node-down
    [Template]    Power Off Node And Longhorn Not Force Delete Terminating Statefulset Pod
        Set Node Down Pod Deletion Policy     do-nothing

Physical Node Down StatefulSet Deletion Policy Delete Deployment Pod
    [Tags]    node-down
    [Template]    Power Off Node And Longhorn Not Force Delete Terminating Statefulset Pod
        Set Node Down Pod Deletion Policy     delete-deployment-pod
```

Reply:
Updated agagin for combine similar test case, and here is the test [result](https://ci.longhorn.io/job/private/job/longhorn-e2e-test/2484/), thank you.

```robot
*** Test Cases ***
Node Down And Longhorn Not Force Delete Terminating StatefulSet Pod
    [Tags]    node-down
    [Template]    Power Off Node And Longhorn Not Force Delete Terminating Statefulset Pod
        do-nothing
        delete-deployment-pod
```